### PR TITLE
NEO-2112 Add filter icon, check mark, menu separator

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[Link to updated components in Deploy Preview](https://deploy-preview-530--neo-react-library-storybook.netlify.app/?path=/story/components-table--advanced-filtering-and-sorting)
[Link to Figma reference](https://www.figma.com/design/o62kJbRQpFq9T0fGwKvCMA/Component-Exploration-FY22---v3?node-id=3649-893596&node-type=frame&t=tetzZlfnGlpWO0Pd-0)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [ ] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

`@avaya-dux/dux-design`: Review

- clicking Date column header, notice menu separator added before Filter Column
- clicking "Filter Column" opens a Column Filter Drawer,
- check some value and click "Apply" to filter
- when filter applied, Filter icon shows in the header, and check mark in front of Filter Column menu item

`@avaya-dux/dux-devs`:  Add icons and separator

Screenshot 2024-09-11 at 1 31 29 PM

## Summary by CodeRabbit

- **New Features**
	- Enhanced filtering functionality in the table header with visual indicators for active filters.
	- Added localization support for filter status with a new translation entry.

- **Improvements**
	- Improved user interface by making the filtering status more apparent and enhancing the menu structure.
	- Updated styles for dropdown headers and filter icons to enhance visual layout.

### CODE REVIEW

**Comment by joe-s-avaya** [comment]
> ## Poem
> > In the header where filters play,
> > A new prop joins the fray!
> > Icons gleam, showing the way,
> > "Filter applied" brightens the day!
> > Hopping through data, oh what a sight,
> > With each click, everything feels right! 🐇✨


This PRs poem is 🔥

File: src/components/Table/TableComponents/TableHeader.tsx [hunk 1]
```diff
@@ -207,18 +214,27 @@ export const TableHeader: TableHeaderComponentType = ({
...
+												)
+											}
+										>
+											{isFiltering && (
+												<Icon
+													icon="check"
+													aria-label={translations.filterApplied}
+												/>
+											)}
+											{translations.filterColumn || "Filter Column"}
```
Comment (joe-s-avaya) [inline src/components/Table/TableComponents/TableHeader.tsx:235]: Is the `||` necessary? I would assume that it already knows the default from `default-data.ts` yes?

**Review by joe-s-avaya — APPROVED** [review]

File: src/components/Table/TableComponents/TableHeader.tsx (hunk 1 above)
Comment (yangAtSpoken) [reply src/components/Table/TableComponents/TableHeader.tsx:235]: good call, removed.

**Comment by matthewAvaya** [comment]
It works as expected. I don't know if this is out of scope but there needs to be more margin, 8px or 0.5 rem, between the label and the icons as shown below. 
Screen Shot 2024-09-11 at 12 36 41 PM

**Comment by Justin-Avaya** [comment]
**Issue 01**
Odd hover state behavior
Open menu from Column Header: Date
Hover across menu items - Hover states display correctly
Hover over Menu Item: Filter Column
Notice both Filter Column and The last Menu Item that was hovered over remain in the hover state

Example


<hr>

**Issue 02**
Can't seem to open menu with keyboard
Example


<hr>

**Issue 03**
Once the Menu is opened (by mouse click) I can navigate up and down the menu with my arrow keys but can not access the Filter Column menu item
Example


<hr>

**Comments**
Hard to tell is these are in/out of scope... Please determine what is in or out and the one's that are identified as out of scope, i will bring them as a parking lot item in our Neo Team Standup meeting.

**Comment by yangAtSpoken** [comment]
> It works as expected. I don't know if this is out of scope but there needs to be more margin, 8px or 0.5 rem, between the label and the icons as shown below. Screen Shot 2024-09-11 at 12 36 41 PM

@matthewAvaya Please check the spacing issue again.

**Comment by yangAtSpoken** [comment]
> **Issue 01** Odd hover state behavior Open menu from Column Header: Date Hover across menu items - Hover states display correctly Hover over Menu Item: Filter Column Notice both Filter Column and The last Menu Item that was hovered over remain in the hover state
> 
> Example  [  ](https://private-user-images.githubusercontent.com/161497909/366615338-3cb5330f-1783-4734-a448-ca2a1e969bcd.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwODcyNDUsIm5iZiI6MTcyNjA4Njk0NSwicGF0aCI6Ii8xNjE0OTc5MDkvMzY2NjE1MzM4LTNjYjUzMzBmLTE3ODMtNDczNC1hNDQ4LWNhMmExZTk2OWJjZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMVQyMDM1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zY2QzY2Y5ZDY0YzNjNTJkZGY2ODViNjRjMmVhMjliZjA2OWVjMWRlYTZmZTVmNGQ0ODdmZmUwZTJjYjJiNjI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UHC9R8oklMniJ0x88-fMVPRAvmW19El4EOZdb4A11n8) [ ](https://private-user-images.githubusercontent.com/161497909/366615338-3cb5330f-1783-4734-a448-ca2a1e969bcd.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwODcyNDUsIm5iZiI6MTcyNjA4Njk0NSwicGF0aCI6Ii8xNjE0OTc5MDkvMzY2NjE1MzM4LTNjYjUzMzBmLTE3ODMtNDczNC1hNDQ4LWNhMmExZTk2OWJjZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMVQyMDM1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zY2QzY2Y5ZDY0YzNjNTJkZGY2ODViNjRjMmVhMjliZjA2OWVjMWRlYTZmZTVmNGQ0ODdmZmUwZTJjYjJiNjI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UHC9R8oklMniJ0x88-fMVPRAvmW19El4EOZdb4A11n8)
> 
> **Issue 02** Can't seem to open menu with keyboard Example  [  ](https://private-user-images.githubusercontent.com/161497909/366616903-181d8834-ccd6-4d0d-8880-00fb4f18836e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwODcyNDUsIm5iZiI6MTcyNjA4Njk0NSwicGF0aCI6Ii8xNjE0OTc5MDkvMzY2NjE2OTAzLTE4MWQ4ODM0LWNjZDYtNGQwZC04ODgwLTAwZmI0ZjE4ODM2ZS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMVQyMDM1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Y2UxYzhiZmJjNzllZWE5ZTZjNWY2MDc3OGQ2MWNlNjdlYmI1ZTk5OTZiNDE5M2VhZGY4NTgwNWQ3YjRjMDJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.8zxKy2fgrbACJkxqQ64kKFt6ZEhnvbhJs0UcVDp1w-g) [ ](https://private-user-images.githubusercontent.com/161497909/366616903-181d8834-ccd6-4d0d-8880-00fb4f18836e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwODcyNDUsIm5iZiI6MTcyNjA4Njk0NSwicGF0aCI6Ii8xNjE0OTc5MDkvMzY2NjE2OTAzLTE4MWQ4ODM0LWNjZDYtNGQwZC04ODgwLTAwZmI0ZjE4ODM2ZS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMVQyMDM1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Y2UxYzhiZmJjNzllZWE5ZTZjNWY2MDc3OGQ2MWNlNjdlYmI1ZTk5OTZiNDE5M2VhZGY4NTgwNWQ3YjRjMDJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.8zxKy2fgrbACJkxqQ64kKFt6ZEhnvbhJs0UcVDp1w-g)
> 
> **Issue 03** Once the Menu is opened (by mouse click) I can navigate up and down the menu with my arrow keys but can not access the Filter Column menu item Example  [  ](https://private-user-images.githubusercontent.com/161497909/366617858-2f4eb67d-23ea-45ad-bd4d-796a307ad991.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwODcyNDUsIm5iZiI6MTcyNjA4Njk0NSwicGF0aCI6Ii8xNjE0OTc5MDkvMzY2NjE3ODU4LTJmNGViNjdkLTIzZWEtNDVhZC1iZDRkLTc5NmEzMDdhZDk5MS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMVQyMDM1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MTYyYTVjOTAxMzVkYjNkYzVhMTAxODAyODhjNmI2OGEzNjY0MzlkNjQ4ZDZhMjNhYzY2OWQ3NTg5ZDliNjhiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Zg_OkTrQ_GGHedrkzJOUUoNF_wZ9jDsTH58EJI2xHa4) [ ](https://private-user-images.githubusercontent.com/161497909/366617858-2f4eb67d-23ea-45ad-bd4d-796a307ad991.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwODcyNDUsIm5iZiI6MTcyNjA4Njk0NSwicGF0aCI6Ii8xNjE0OTc5MDkvMzY2NjE3ODU4LTJmNGViNjdkLTIzZWEtNDVhZC1iZDRkLTc5NmEzMDdhZDk5MS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMVQyMDM1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MTYyYTVjOTAxMzVkYjNkYzVhMTAxODAyODhjNmI2OGEzNjY0MzlkNjQ4ZDZhMjNhYzY2OWQ3NTg5ZDliNjhiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Zg_OkTrQ_GGHedrkzJOUUoNF_wZ9jDsTH58EJI2xHa4)
> 
> **Comments** Hard to tell if these are in/out of scope... Please determine what is in or out and those that are identified as out of scope, I will bring them as a parking lot item in our Neo Team Standup meeting.

@Justin-Avaya These issues are out of the scope of this PR.  And I created [this pr](https://avaya.atlassian.net/browse/NEO-2385) to fix it.  So please approve this if no other issues.

**Review by matthewAvaya — APPROVED** [review]

**Review by Justin-Avaya — APPROVED** [review]

**Pull request merged**